### PR TITLE
Don't build `std` on `no_std` targets

### PR DESCRIPTION
**Pull request by robertbastian**
https://github.com/rust-lang/rust/issues/126992

### CODE REVIEW

File: ffi/npm/Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ lib/index.mjs:
 	rustup component add rust-src --toolchain ${PINNED_CI_NIGHTLY}
 	RUSTFLAGS="-Cpanic=abort -Copt-level=s -C link-args=-zstack-size=${WASM_STACK_SIZE} -Clto -Cembed-bitcode" \
 	cargo +${PINNED_CI_NIGHTLY} rustc \
-		-Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort \
```
Comment (Manishearth) [inline ffi/npm/Makefile:23]: question: why remove panic_abort here and below?

File: tools/make/ffi.toml [hunk 2]
```diff
@@ -71,7 +71,7 @@ command = "cargo"
 args = ["check", "--package", "icu_freertos",
         "--target", "thumbv8m.main-none-eabihf",
         "--no-default-features", "--features=wearos",
-        "-Zbuild-std=std,panic_abort", "-Zbuild-std-features=panic_immediate_abort"]
+        "-Zbuild-std=core,alloc", "-Zbuild-std-features=panic_immediate_abort"]
```
Comment (Manishearth) [inline tools/make/ffi.toml:74]: this should still be panic_abort, yes?

**Review by Manishearth — CHANGES_REQUESTED** [review]
Why are we also removing panic_abort?

File: ffi/npm/Makefile (hunk 1 above)
Comment (robertbastian) [reply ffi/npm/Makefile:23]: It's not a valid argument: https://doc.rust-lang.org/cargo/reference/unstable.html#build-std:~:text=The%20value%20here%20is%20a%20comma%2Dseparated%20list%20of%20standard%20library%20crates%20to%20build.

File: tools/make/ffi.toml (hunk 2 above)
Comment (bjorn3) [reply tools/make/ffi.toml:74]: The panic_abort crate is only used by libstd.

File: ffi/npm/Makefile (hunk 1 above)
Comment (bjorn3) [reply ffi/npm/Makefile:23]: panic_abort certainly is a valid argument. It is one of the crates that is part of the standard library. I guess it works without by accident when panic_immediate_abort is being used, but when it isn't used `-Cpanic=abort` requires `panic_abort` to be used in the `-Zbuild-std` list.

Comment (robertbastian) [reply ffi/npm/Makefile:23]: Thanks, this isn't documented anywhere

**Review by Manishearth — APPROVED** [review]

**Pull request merged**